addons/mail/static/src/models/tracking_value_item.js

```diff
@@ -0,0 +1,111 @@
+                 * possibilities and they are not given in this context.
+                 */
+                case 'char':
+                case 'many2one':
+                case 'selection':
+                    return format.char(this.value);
+                case 'date':
+                    if (this.value) {
+                        return format.date(moment.utc(this.value));
+                    }
+                    return format.date(this.value);
+                case 'datetime':
+                    if (this.value) {
+                        return format.datetime(moment.utc(this.value));
+                    }
+                    return format.datetime(this.value);
+                case 'float':
+                    return format.float(this.value);
+                case 'integer':
+                    return format.integer(this.value);
+                case 'text':
+                    return format.text(this.value);
+                case 'monetary':
+                    return format.monetary(this.value, undefined, {
+                        currency: this.currencyId
+                            ? this.env.session.currencies[this.currencyId]
+                            : undefined,
+                        forceString: true,
+                    });
+                default :
+                    return this.value;
+            }
+        },
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeValueAsString() {
+            return this.formattedValue || this.env._t("None");
+        },
+    },
+    fields: {
+        /**
+         * Used when the currency changes as the tracking value. This only makes sense for field of type monetary.
+         */
+        currencyId: attr(),
+        /**
+         * Indicates the type of the tracking value item.
+         * The supported types are: boolean, char, many2one, selection, date, datetime, float, integer, text, monetary.
+         */
+        fieldType: attr(),
+        /**
+         * The formatted value of the tracking value item.
+         */
+        formattedValue: attr({
+            compute: '_computeFormattedValue',
+        }),
+        /**
+         * States the tracking value displaying the item.
+         * Generally, it is mutually exclusive with trackingValueAsOldValue.
+         */
+        trackingValueAsNewValue: one('TrackingValue', {
+            inverse: 'newValue',
+            readonly: true,
+        }),
+        /**
+         * States the tracking value displaying the item.
+         * Generally, it is mutually exclusive with trackingValueAsNewValue.
+         */
+        trackingValueAsOldValue: one('TrackingValue', {
+            inverse: 'oldValue',
+            readonly: true,
+        }),
+        /**
+         * The original value of the tracking value item.
+         */
+        value: attr(),
+        valueAsString: attr({
+            compute: '_computeValueAsString',
+        }),
```

Comment:
confusing name, maybe just `formattedValueOrNone`?

And again I think it would be better if `formattedValue` included `None` by default, and the check for emptiness be done based on just value instead if possible.

Reply:
But if you check the original value, it still can be an empty string after formatted, which is what we want to get rid of.

Reply:
what I can expect is a string with spaces like `   ` and datetime but not in the right format, the formatter will return '' as result.